MantidPlot/src/ApplicationWindow.cpp

```diff
@@ -14041,9 +14042,30 @@ void ApplicationWindow::updateRecentFilesList(QString fname) {
     recentFiles.pop_back();
 
   recentFilesMenu->clear();
-  for (int i = 0; i < (int)recentFiles.size(); i++)
-    recentFilesMenu->insertItem("&" + QString::number(i + 1) + " " +
-                                recentFiles[i]);
+  for (int i = 0; i < (int)recentFiles.size(); i++) {
+    QMenuItem *mi = new QMenuItem;
+
+    try {
+      Mantid::API::MultipleFileProperty mfp("tester");
+      mfp.setValue(recentFiles[i].toStdString());
+      const std::vector<std::string> files =
+          Mantid::API::MultipleFileProperty::flattenFileNames(mfp());
+      if (files.size() == 1) {
+        mi->setText("&" + QString::number(i + 1) + " " + recentFiles[i]);
+      } else {
+        std::ostringstream ostr;
+        ostr << "&" << QString::number(i + 1).toStdString() << " " << files[0]
+             << " && " << files.size() - 1 << " more";
+        mi->setText(QString::fromStdString(ostr.str()));
+      }
+    } catch (std::runtime_error &ex) {
+      UNUSED_ARG(ex);
```

Comment:
Consider not giving the exception a name (`catch (std::runtime_error &)`) to avoid the unused variable warning / having to use the macro workaround.